CMakeLists.txt

```diff
@@ -13,20 +14,26 @@ endif()
 # Create Mbed OS library
 add_library(mbed-os OBJECT)
 
+include(${MBED_ROOT}/cmake/util.cmake)
+
 include(${MBED_CONFIG_PATH}/mbed_config.cmake)
 
 include(${MBED_ROOT}/cmake/core.cmake)
 mbed_set_cpu_core_options(mbed-os ${MBED_TOOLCHAIN})
 
-include(${MBED_ROOT}/cmake/toolchain.cmake)
+# Set the toolchain file, do not include it as it's included by CMake on system level
+if(NOT CMAKE_TOOLCHAIN_FILE)
+    set(CMAKE_TOOLCHAIN_FILE "${MBED_ROOT}/cmake/toolchain.cmake" CACHE INTERNAL "")
+endif()
+
+include(${MBED_ROOT}/cmake/toolchains/${MBED_TOOLCHAIN}.cmake)
+enable_language(C CXX ASM)
```

Comment:
What do you think about adding `project(mbed-os LANGUAGES C CXX ASM)` instead towards the top of the file?

Reply:
I'll fix that in the new PR (separate from response files)